chromadb/test/property/strategies.py

```diff
@@ -39,46 +39,130 @@ def collections(draw) -> Collection:
     """Strategy to generate a set of collections"""
 
     # name = draw(st.from_regex(coll_name_re))
-    name = draw(st.one_of(st.from_regex(_coll_name_re)))
+    name = draw(st.one_of(st.from_regex(_collection_name_re)))
     hypothesis.assume(not _ipv4_address_re.match(name))
     hypothesis.assume(not _two_periods_re.search(name))
 
-    return {"name": name, "metadata": draw(metadata)}
+    return {"name": name, "metadata": draw(collection_metadata)}
+
+
+def one_or_both(strategy_a, strategy_b):
+    return st.one_of(
+        st.tuples(strategy_a, strategy_b),
+        st.tuples(strategy_a, st.none()),
+        st.tuples(st.none(), strategy_b),
+    )
+
+
+@st.composite
+def unique_ids_strategy(draw, count: int):
+
+    ratio = 20
+    strs = count // ratio
+
+    str_results = draw(
+        st.lists(st.text(min_size=1, max_size=64), min_size=strs, max_size=strs, unique=True)
+    )
+
+    # Rotate selections from between the two lists
+    results = []
+    for i in range(count):
+        if i % ratio == 0 and len(str_results) > 0:
+            results.append(str_results.pop())
+        else:
+            results.append(str(draw(st.uuids())))
+
+    return results
+
+
+float_types = [np.float16, np.float32, np.float64]
+int_types = [np.int16, np.int32, np.int64]
+
+
+def embeddings_strategy(dim: int, count: int, dtype: np.dtype):
+    return npst.arrays(
+        dtype=dtype,
+        shape=(count, dim),
+        # TODO: It should be possible to deal with NaN and inf values
+        # TODO: It should be possible to deal with redundant embeddings
+        elements=st.floats(
+            allow_nan=False,
+            allow_infinity=False,
+            width=np.dtype(dtype).itemsize * 8,
+        )
+        if dtype in float_types
+        else st.integers(min_value=np.iinfo(dtype).min, max_value=np.iinfo(dtype).max),
+        unique=True,
+    )
+
+
+# TODO: Use a hypothesis strategy while maintaining embedding uniqueness
+#       Or handle duplicate embeddings within a known epsilon
+def create_embeddings(dim: int, count: int, dtype: np.dtype):
+    return np.random.uniform(
```

Comment:
Any specific reason we're using raw `np.random` instead of the pre-built Hypothesis Numpy wrapper in `hypothesis.extra.numpy`? Did that not work for us for some reason?